### PR TITLE
Improve coupons discounts description

### PR DESCRIPTION
**Pull request by VladimirBerdnik**

### CODE REVIEW

File: spec/components/schemas/Coupon/Discounts/percent.yaml [hunk 1]
```diff
@@ -9,5 +9,6 @@ allOf:
     properties:
       value:
         description: Discount percent
-        type: number
-        format: double
+        type: integer
+        minimum: 1
```
Comment (slavcodev) [inline spec/components/schemas/Coupon/Discounts/percent.yaml]:
```suggestion
        minimum: 0
```

File: spec/components/schemas/Coupon/Discounts/fixed.yaml [hunk 2]
```diff
@@ -12,6 +12,8 @@ allOf:
         description: Discount amount
         type: number
         format: double
+        minimum: 0
+        exclusiveMinimum: true
```
Comment (slavcodev) [inline spec/components/schemas/Coupon/Discounts/fixed.yaml]:
```suggestion
```

File: spec/components/schemas/Coupon/Discounts/percent.yaml (hunk 1 above)
Comment (VladimirBerdnik) [reply spec/components/schemas/Coupon/Discounts/percent.yaml]: @slavcodev, does it have sence to have such coupons?

File: spec/components/schemas/Coupon/Discounts/fixed.yaml (hunk 2 above)
Comment (adamaltman) [reply spec/components/schemas/Coupon/Discounts/fixed.yaml]: Why would you remove this?  It makes a lot of sense.

File: spec/components/schemas/Coupon/Discounts/percent.yaml (hunk 1 above)
Comment (adamaltman) [reply spec/components/schemas/Coupon/Discounts/percent.yaml]: No, I disagree with the suggestion.

File: spec/components/schemas/Coupon/Discounts/fixed.yaml (hunk 2 above)
Comment (slavcodev) [reply spec/components/schemas/Coupon/Discounts/fixed.yaml]: I don't see why it cannot be 0. But, I think I understand your point of view.

File: spec/components/schemas/Coupon/Discounts/percent.yaml (hunk 1 above)
Comment (slavcodev) [reply spec/components/schemas/Coupon/Discounts/percent.yaml]: I don't see why we should disallow this case. It does not break any logic of application.

**Comment by VladimirBerdnik** [comment]
So, no notices on this PR? can it be approved?

**Comment by VladimirBerdnik** [comment]
@slavcodev , @adamaltman , do we have agreement on this PR? :)

File: spec/components/schemas/Coupon/Discounts/percent.yaml [hunk 3]
```diff
@@ -9,5 +9,6 @@ allOf:
     properties:
       value:
         description: Discount percent
-        type: number
-        format: double
+        type: integer
```
Comment (slavcodev) [inline spec/components/schemas/Coupon/Discounts/percent.yaml]: Is the percentage discount maybe only integer?

File: spec/components/schemas/Coupon/Discounts/percent.yaml [hunk 4]
```diff
@@ -9,5 +9,6 @@ allOf:
...
+        minimum: 1
+        maximum: 100
```
Comment (slavcodev) [inline spec/components/schemas/Coupon/Discounts/percent.yaml]: Just curios is the discount 200% posible?

File: spec/components/schemas/Coupon/Discounts/percent.yaml (hunk 3 above)
Comment (VladimirBerdnik) [reply spec/components/schemas/Coupon/Discounts/percent.yaml]: I took this restriction from backend

Comment (slavcodev) [reply spec/components/schemas/Coupon/Discounts/percent.yaml]: I think we need to fix backend.

Comment (VladimirBerdnik) [reply spec/components/schemas/Coupon/Discounts/percent.yaml]: I actually can't imagine a use case for float percentage for disount

Comment (VladimirBerdnik) [reply spec/components/schemas/Coupon/Discounts/percent.yaml]: It wasn't allowed yet and looks like we have no any requests from our clients to have such possibility

**Comment by 2e3s** [comment]
I wonder if the maximum percentage is <100

**Comment by VladimirBerdnik** [comment]
> I wonder if the maximum percentage is <100

you mean should be less than 100? I think having 100% discount coupon has sense

**Comment by 2e3s** [comment]
Oh, yes. I mean <=100%. 120% discount is a strange thing, credits should be added differently.

**Comment by 2e3s** [comment]
Sorry, I missed that it's already there

**Pull request merged**